common/src/main/java/com/microsoft/identity/common/exception/BaseException.java

```diff
@@ -42,6 +42,8 @@
 
     private String mErrorCode;
 
+    private String mSubErrorCode;
```

Comment:
One thing that jumps out to me here is the somewhat confusing API `BaseException` has developed - looking at this class we see `errorCode` and `subErrorCode` and would naturally assume "well, those must be error response payloads". In actuality though, we populate `errorCode` with our own non-service related codes too ([example](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/common/src/main/java/com/microsoft/identity/common/exception/ArgumentException.java#L39)).

Just thinking out loud.... it would be helpful to have our errors _and_ their origins be clear. 

For the purposes of this PR, would it make sense to beef-up `ServiceException` to provide access to error and suberror payloads and use that to signal instances where popping UI won't resolve the issue? If yes, could we roll specific errors (which may been returned to us by the service in subErrors) up to a series of single error codes we surface? `WPJ_DEVICE_IS_DISABLED` etc etc?

Reply:
Feedback taken. Will push out a change regarding the error code series in ad_accounts at some point